### PR TITLE
Refactoring Bloated Code in C#

### PR DESCRIPTION
**Pull request by zanbor**

### CODE REVIEW

**Comment by karimkod** [comment]
retest this please

File: csharp-refactoring/RefactoringCodeBloaters/Tests/Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (karimkod) [inline csharp-refactoring/RefactoringCodeBloaters/Tests/Tests.csproj:4]: We should target net7.0

File: csharp-refactoring/RefactoringCodeBloaters/RefactoringCodeBloaters/RefactoringCodeBloaters.csproj [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (karimkod) [inline csharp-refactoring/RefactoringCodeBloaters/RefactoringCodeBloaters/RefactoringCodeBloaters.csproj:5]: We should target net7.0

File: csharp-refactoring/RefactoringCodeBloaters/Tests/CalculationServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using RefactoringCodeBloaters.LongMethod.Correct;
+
+namespace Tests
+{
+    [TestClass]
+    public class CalculationServiceTests
```
Comment (karimkod) [inline csharp-refactoring/RefactoringCodeBloaters/Tests/CalculationServiceTests.cs:7]: Unit tests file and classes should end with UnitTest e.g "CalculationServiceUnitTest" and "CalculationServiceUnitTest.cs" respectively. This applies to all unit tests files in the test project.

File: csharp-refactoring/RefactoringCodeBloaters/Tests/CalculationServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        private readonly CalculationService calculationService;
+        public CalculationServiceTests()
+        {
+            calculationService = new CalculationService();
+        }
+
+        [TestMethod]
+        public void CalculateDiscount_LessThan250_ReturnsOriginalPrice()
```
Comment (karimkod) [inline csharp-refactoring/RefactoringCodeBloaters/Tests/CalculationServiceTests.cs:16]: As stated in the guidelines, - 
test should follow the BDD convention: GivenX_WhenY_ThenZ
- the given part is optional, but the other two are not
- also, the delimiter (underline) should only be used between these sections, and not anywhere else
GivenPriceLessThan250_WhenCalculatingPriceAfterDiscount_ThenReturnOriginalPrice()

This applies to all Test methods as well.

**Review by karimkod — CHANGES_REQUESTED** [review]

**Comment by CodeMazeBlog** [comment]
retest this please

File: csharp-refactoring/README.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+## Refactoring in C#
```
Comment (karimkod) [inline csharp-refactoring/README.md:1]: README.md should be committed separately than the project. So, just: 
- Remove the README.md from the project.
- Commit the changes. 
- And then create a separate commit adding the README.md

**Review by karimkod — CHANGES_REQUESTED** [review]

**Comment by zanbor** [comment]
retest this please

File: csharp-refactoring/README.md (hunk 5 above)
Comment (karimkod) [reply csharp-refactoring/README.md:1]: @zanbor can you please remove the README.md completely from this PR. The team will add it later. 
Sorry, I was confused with what exactly should be done. 
This is the reason the build is failing. Thank you

**Review by karimkod — CHANGES_REQUESTED** [review]

**Pull request merged**